plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/textdocument/TextDocumentServiceHandler.kt

```diff
@@ -141,4 +155,28 @@ class TextDocumentServiceHandler(
             }
         }
     }
+
+    private fun realTimeEdit(event: DocumentEvent) {
+        AmazonQLspService.executeIfRunning(project) { languageServer ->
+            pluginAwareExecuteOnPooledThread {
+                val vFile = FileDocumentManager.getInstance().getFile(event.document) ?: return@pluginAwareExecuteOnPooledThread
+                toUriString(vFile)?.let { uri ->
```

Comment:
why nullable

Reply:
toUriString() is an existing method that will return nullable strings in rare cases
```
fun toUriString(virtualFile: VirtualFile): String? {
        val protocol = virtualFile.fileSystem.protocol
        val uri = when (protocol) {
            "jar" -> VfsUtilCore.convertToURL(virtualFile.url)?.toExternalForm()
            "jrt" -> virtualFile.url
            else -> toUri(VfsUtilCore.virtualToIoFile(virtualFile)).toASCIIString()
        } ?: return null

        return if (virtualFile.isDirectory) {
            uri.trimEnd('/', '\\')
        } else {
            uri
        }
    }
```